src/IJulia.jl

```diff
@@ -211,6 +211,50 @@ a subset of cell inputs/outputs to clear.
 clear_history
 
 #######################################################################
+# methods to print history or any subset thereof
+function print_history(io::IO, indices)
+    for n in indices
```

Comment:
You can just do `for n in intersect(indices, 1:IJulia.n)` and remove most of the special cases below